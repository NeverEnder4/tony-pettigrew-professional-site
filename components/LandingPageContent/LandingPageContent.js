import Link from 'next/link';

import HighLightLink from '../../components/HighLightLink/HighLightLink';

import './LandingPageContent.scss';

const LandingPageContent = () => (
  <React.Fragment>
    <div className="headline-scaffold">
      <div className="headline-row">
        <h1 className="headline">I build</h1>
      </div>
      <div className="headline-row">
        <h1 className="headline">websites</h1>
      </div>
      <div className="headline-row last">
        <h1 className="headline">that</h1>
        <img
          className="logo"
          src="/static/svg/champagne-logo.svg"
          alt="champagne bottle logo"
        />
      </div>
    </div>
    <div className="landing-lower-section">
      <div className="vertical-line">
        <nav className="landing-nav">
          <ul>
            <li className="landing-nav-item">
              <Link href="/biographique">
                <a>biographique</a>
              </Link>
            </li>
            <li className="landing-nav-item">
              <Link href="/works">
                <a>works</a>
              </Link>
            </li>
            <li className="landing-nav-item">
              <Link href="/thoughts">
                <a>thoughts</a>
              </Link>
            </li>
            <li className="landing-nav-item">
              <Link href="mailto:apettigrew.wsdev@gmail.com">
                <a>tacoboutit</a>
              </Link>
            </li>
          </ul>
        </nav>

        <span>Pull Me</span>
        <img
          className="sphere-icon"
          src="/static/svg/sphere.svg"
          alt="sphere"
        />
      </div>
    </div>
    <footer className="social-links">
      <HighLightLink href="#" text="Linkedin" />
      <HighLightLink href="#" text="Github" />
      <HighLightLink href="#" text="Twitter" />
    </footer>
  </React.Fragment>
);

export default LandingPageContent;
