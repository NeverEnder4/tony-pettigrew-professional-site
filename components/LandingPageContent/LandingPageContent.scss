@import '../../scss/abstractions/abstractions';

// upper landing page, headline & logo
.headline-scaffold {
  width: 100%;
  min-width: 300px;
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  @media (min-width: 375px) {
    width: 355px;
  }
  @media (min-width: 1280px) {
    width: 433px;
  }
}

.headline-row {
  border-left: $landing-border;
  border-top: $landing-border;
  border-right: $landing-border;
  display: flex;
  &:last-child {
    position: relative;
    border-bottom: $landing-border;
  }
}

.headline {
  font-size: 6rem;
  margin: 2px 0 2px 5px;
  @media (min-width: 375px) {
    font-size: $font-size-xl;
  }
  @media (min-width: 1280px) {
    font-size: $font-size-xxl;
    margin: 2px 0 2px 7px;
  }
}

.logo {
  width: 120px;
  position: absolute;
  left: 45vw;
  top: -22px;
  @media (min-width: 375px) {
    width: 140px;
    left: 170px;
    top: -25px;
  }
  @media (min-width: 1280px) {
    width: 175px;
    left: 220px;
    top: -32px;
  }
}

// lower landing page, menu & line
.vertical-line {
  position: relative;
  border-right: $landing-border;
  width: 50%;
  height: 43vh;
}

// navigation

.landing-nav {
  position: relative;
  top: 25%;
  padding-right: 10px;
}

.landing-nav ul {
  list-style-type: none;
  padding: 0;
  margin: 0;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
}

.landing-nav-item a {
  text-decoration: none;
  color: $color-black;
  font-size: $font-size-sm;
  transition: color 100ms cubic-bezier(0.47, 0, 0.745, 0.715);
  @media (min-width: 375px) {
    font-size: $font-size-md;
  }
  @media (min-width: 1280px) {
    font-size: $font-size-xm;
  }
}

.landing-nav-item a:hover,
.landing-nav-item a:focus {
  color: $color-white;
}

// sphere image and pull me span
.sphere-icon {
  width: 55px;
  position: absolute;
  bottom: -43px;
  right: -29px;
  @media (min-width: 1280px) {
    width: 70px;
    bottom: -55px;
    right: -36.8px;
  }
}

// pull me text
.vertical-line span {
  font-size: 1rem;
  text-transform: uppercase;
  transform: rotate(-90deg);
  position: absolute;
  bottom: 6vh;
  right: -34px;
  @media (min-width: 375px) {
    font-size: $font-size-xs;
    bottom: 4vh;
    right: -39px;
  }
}

// social links footer
.social-links {
  display: none;
  position: absolute;
  width: 37%;
  right: 2vw;
  bottom: 3vh;
  justify-content: space-between;
  @media (min-width: 1280px) {
    display: flex;
  }
}
