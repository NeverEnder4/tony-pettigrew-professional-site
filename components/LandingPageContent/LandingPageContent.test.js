import renderer from 'react-test-renderer';
import LandingPageContent from './LandingPageContent';

it('renders and matches the current snapshot', () => {
  expect(renderer.create(<LandingPageContent />).toJSON()).toMatchSnapshot();
});
