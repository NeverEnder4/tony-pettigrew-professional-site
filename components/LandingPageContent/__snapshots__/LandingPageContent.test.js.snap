// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and matches the current snapshot 1`] = `
Array [
  <div
    className="headline-scaffold"
  >
    <div
      className="headline-row"
    >
      <h1
        className="headline"
      >
        I build
      </h1>
    </div>
    <div
      className="headline-row"
    >
      <h1
        className="headline"
      >
        websites
      </h1>
    </div>
    <div
      className="headline-row last"
    >
      <h1
        className="headline"
      >
        that
      </h1>
      <img
        alt="champagne bottle logo"
        className="logo"
        src="/static/svg/champagne-logo.svg"
      />
    </div>
  </div>,
  <div
    className="landing-lower-section"
  >
    <div
      className="vertical-line"
    >
      <nav
        className="landing-nav"
      >
        <ul>
          <li
            className="landing-nav-item"
          >
            <a
              href="/biographique"
              onClick={[Function]}
            >
              biographique
            </a>
          </li>
          <li
            className="landing-nav-item"
          >
            <a
              href="/works"
              onClick={[Function]}
            >
              works
            </a>
          </li>
          <li
            className="landing-nav-item"
          >
            <a
              href="/thoughts"
              onClick={[Function]}
            >
              thoughts
            </a>
          </li>
          <li
            className="landing-nav-item"
          >
            <a
              href="mailto:apettigrew.wsdev@gmail.com"
              onClick={[Function]}
            >
              tacoboutit
            </a>
          </li>
        </ul>
      </nav>
      <span>
        Pull Me
      </span>
      <img
        alt="sphere"
        className="sphere-icon"
        src="/static/svg/sphere.svg"
      />
    </div>
  </div>,
  <footer
    className="social-links"
  >
    <a
      className="high-light-link"
      href="#"
      rel="noreferrer"
    >
      Linkedin
    </a>
    <a
      className="high-light-link"
      href="#"
      rel="noreferrer"
    >
      Github
    </a>
    <a
      className="high-light-link"
      href="#"
      rel="noreferrer"
    >
      Twitter
    </a>
  </footer>,
]
`;
